app/main.py

```diff
@@ -1,2 +1,66 @@
+from typing import Any, Optional
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length: int = 0
+        self.capacity: int = 8
+        self.hash_table: list[Optional[tuple]] = [None] * self.capacity
+
+    def _get_key_value(self, key: Any) -> int | None:
+        index = hash(key) % self.capacity
+        original_index = index
+
+        while self.hash_table[index] is not None:
+            if self.hash_table[index][0] == key:
+                return self.hash_table[index][2]
+            index = (index + 1) % self.capacity
+
+            if index == original_index:
+                break
+
+        return None
+
+    def _set_key_value(self, key: Any, value: Any) -> int | None:
+        self._check_extend_memory()
+
+        index = hash(key) % self.capacity
+        original_index = index
+
+        while self.hash_table[index] is not None:
+            if self.hash_table[index][0] == key:
+                self.hash_table[index] = (key, hash(key), value)
+                return
+            index = (index + 1) % self.capacity
+
+            if index == original_index:
+                break
+
+        self.hash_table[index] = (key, hash(key), value)
+        self.length += 1
+
+    def __getitem__(self, key: Any) -> Any:
+        value = self._get_key_value(key)
+        if value is None:
```

Comment:
```suggestion
        if not value:
```

Reply:
when I use:

    if not value:

instead:

    if value is None:

tests crash, when value == 0

I think it is because bool(0) == False,  bool(None) == False and bool(other type) == True.